daemon/oci_linux.go

```diff
@@ -666,7 +666,7 @@ func withMounts(daemon *Daemon, daemonCfg *configStore, c *container.Container,
 
 		// TODO: until a kernel/mount solution exists for handling remount in a user namespace,
 		// we must clear the readonly flag for the cgroups mount (@mrunalp concurs)
-		if uidMap := daemon.idMapping.UIDMaps; uidMap != nil || c.HostConfig.Privileged {
+		if uidMap := daemon.idMapping.UIDMaps; uidMap != nil || c.HostConfig.Privileged || c.Cgroups == "rw" {
```

Comment:
"rw" should be prohibited on cgroup v1 hosts, as the container can be escaped via a write access to the device controller

Reply:
good catch. Will need to cover that situation.

Reply:
To put that another way, IMO the v1 concerns are more of a documentation problem than something we should block in the code.

Reply:
> this has security implications on the current v2 implementation also, so why block it for v1?

Because the issue is more critical on v1.

Reply:
https://github.com/moby/moby/pull/42043#issuecomment-781990747

and sure, but `--privileged` does it, so this is more about having a way to get that bit of `--privileged` without the full hammer (https://github.com/moby/moby/issues/47663) than it is about babysitting a user who's explicitly asking for an insecure (or potentially insecure) configuration -- all the `--security-opt` settings are "user accepts the risk" type peeling back of the security boundary of the container, and it's already up to users to determine how that affects them, their use case, and their security model (and v1 is on the way out completely anyhow)